kernel/src/codec_manager.cpp

```diff
@@ -59,4 +73,31 @@ status_t getCodec(std::string name, CodecInfo &codec_info)
     }
 
     return 1;
+}
+
+//fun what search files whith need extension
+std::list <std::string> recursiveDLLSearch(const std::string& search_path, const std::string& extension) {
+
+	std::list <std::string> list_of_path_to_dll; // create local list? what contained pathes
+
+	try {
+		for (auto& p : std::filesystem::recursive_directory_iterator(search_path)) {//cycle what run in all folders
+
+			if (!std::filesystem::is_regular_file(p.status()))//check if isn`t directory, if it not file we are continue our cycle
+				continue;
+
+			std::string name = p.path().filename().u8string();//take name of current file
+			bool match = !name.compare(name.size() - extension.size(), extension.size(), extension);//check if file has extension what we need
+
+			if (!match)
+				continue;
+			list_of_path_to_dll.push_back(p.path().u8string());//write path to file into list
+		}
+	}
+	// catch error and return what error we have(like unfound file
+	catch (std::exception & e) {
+		std::cout << "Error: " << e.what() << '\n';

```

Comment:
What type of error you are catching here? Is there need to pass every general exception to the user stdout?

Reply:
Here I catch an error when the bin folder does not exist in the root. There I use cout, because I want to out error on screen, if we will have GUI, we can to call function or anithing else, what say to user, what we can`t find need folder. Now it's just a plug